src/acceptor.rs

```diff
@@ -0,0 +1,139 @@
+use core::task::{Context, Poll};
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::sync::Arc;
+
+use futures_util::ready;
+use hyper::server::{
+    accept::Accept,
+    conn::{AddrIncoming, AddrStream},
+};
+use rustls::ServerConfig;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+mod builder;
+pub use builder::AcceptorBuilder;
+use builder::WantsTlsConfig;
+
+enum State {
+    Handshaking(tokio_rustls::Accept<AddrStream>),
+    Streaming(tokio_rustls::server::TlsStream<AddrStream>),
+}
+
+// tokio_rustls::server::TlsStream doesn't expose constructor methods,
```

Comment:
I've been taking a little look at this, and it seems quite tough to implement or require a partial rewrite. The TlsStream from the server example (this here) does the handshaking part which the TlsStream from `tokio-rustls` doesn't have, so that would change it (and I guess break other uses of it).

Perhaps it's best to tackle that in a different PR? it'll probably require a bit more discussion

Reply:
Ah sorry, sure, let's skip that!